XamarinCommunityToolkit/Extensions/iOS/ImageResourceExtension.ios.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using System.IO;
+
+using UIKit;
+using NativeImage = UIKit.UIImage;
+
+using Xamarin.Forms;
+using Xamarin.Forms.Internals;
+using Xamarin.Forms.Platform.iOS;
+using System;
+
+namespace Xamarin.CommunityToolkit.Extensions.iOS
+{
+	public static class ImageResourceExtention
+	{
+		internal static async Task<NativeImage> GetNativeImageAsync(this ImageSource source, CancellationToken cancellationToken = default(CancellationToken))
+		{
+			if (source == null || source.IsEmpty)
+				return null;
+
+			var handler = Forms.Internals.Registrar.Registered.GetHandlerForObject<IImageSourceHandler>(source);
+			if (handler == null)
+				return null;
+
+			try
+			{
+#if __MOBILE__
```

Comment:
This file doesn't compile to MacOS, can you remove this?

Reply:
Renamed the file to compile for MacOS.